File: solo/easy/skynet_the_chasm.ml
let road = int_of_string (input_line stdin) in (* the length of the road before the gap. *)
let gap  = int_of_string (input_line stdin) in (* the length of the gap. *)
let platform = int_of_string (input_line stdin) in (* the length of the landing platform. *)

(* game loop *)
while true do
    let speed = int_of_string (input_line stdin) in (* the motorbike's speed. *)
    let coordx = int_of_string (input_line stdin) in (* the position on the road of the motorbike. *)

    print_endline "SPEED"; (* A single line containing one of 4 keywords: SPEED, SLOW, JUMP, WAIT. *)
done;

File: solo/easy/javascript/skynet_the_chasm.js
var R = parseInt(readline()); // the length of the road before the gap.
var G = parseInt(readline()); // the length of the gap.
var L = parseInt(readline()); // the length of the landing platform.

var fl_s = 0;

// game loop
while (true) {
    var S = parseInt(readline()); // the motorbike's speed.
    var X = parseInt(readline()); // the position on the road of the motorbike.

    if (S > G + 1) {
        fl_s = 0;
    } else {
        fl_s = 1;
    }

    if (X < R && fl_s == 1) {
        if (S == G + 1) {
            if (X == R - 1) {
                print('JUMP');
            } else {
                print('WAIT');
            }
        } else {
            print('SPEED');
        }
    } else {
        print('SLOW');
    } // A single line containing one of 4 keywords: SPEED, SLOW, JUMP, WAIT.
}
